### PR TITLE
feat(Pointer): offset playAreaCursor to hint at relative player position

### PR DESCRIPTION
**Pull request by sdouglas**
Add offset to playAreaCursor relative to hmd position, per discussion in issue #173

### CODE REVIEW

File: Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs [hunk 1]
```diff
@@ -97,7 +98,12 @@ protected virtual void InitPointer()
 
         protected virtual void SetPlayAreaCursorTransform(Vector3 destination)
         {
-            playAreaCursor.transform.position = destination;
+            playArea = GameObject.FindObjectOfType<SteamVR_PlayArea>();
```
Comment (thestonefox) [inline Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs]: I would recommend doing this assignment in the Start() or Awake() method to reduce the need for it to be called every time the cursor is drawn (reduces overhead on a costly Find call).

If it doesn't work in Start() or Awake() then maybe do a check to see if it's already been set:

```
if (playArea == null) {
  playArea = GameObject.FindObjectOfType<SteamVR_PlayArea>();
}
```

File: Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs [hunk 2]
```diff
@@ -97,7 +98,12 @@ protected virtual void InitPointer()
...
+            hmdCamera = GameObject.FindObjectOfType<SteamVR_Camera>();
```
Comment (thestonefox) [inline Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs]: Use the VRTK.DeviceFinder.HeadsetTransform() method for getting the hmd as it handles the differences between unity 5.3 and 5.4

the method is: https://github.com/thestonefox/SteamVR_Unity_Toolkit/blob/master/Assets/SteamVR_Unity_Toolkit/Scripts/Helper/DeviceFinder.cs#L35

so this line would be (with a null check as well for performance):

```
if (hmdCamera == null) {
  hmdCamera = DeviceFinder.HeadsetTransform();
}
```

**Comment by thestonefox** [comment]
Thanks for the PR, looks good! I've just left a couple of points to look into :)

**Comment by sdouglas** [comment]
Great suggestions, thanks! I'll revise.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 0681644d8748d4858604860fd54524a4152cf618
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

**Comment by thestonefox** [comment]
Close to being mergeable now :)

Just need to fix the commit message so it adheres to the gitcop rules.

Your original commit message would be fine, would be good to also include a longer description too.

**Comment by sdouglas** [comment]
Ok, I revised my second commit message that violated your gitcop rule and used `git push --force` to overwrite it. But I realize now that it might have been cleaner to repeatedly revise the original commit, since you don't necessarily want to merge a bunch of incremental commits from my branch as I figure out what to do. Or do you mind having those commits as long as they conform to the formatting rules?

**Comment by thestonefox** [comment]
I'd squash your changes so there is only 1 commit.

May be easier to just re-implement your changes in a new commit if you're not used to squashing git commits (it's only 3 lines really).

There's no need to have the history of the commit as the latest commit is pretty much the work you want to commit.

File: Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs [hunk 3]
```diff
@@ -28,6 +28,7 @@ public abstract class VRTK_WorldPointer : VRTK_DestinationMarker
         protected bool playAreaCursorCollided = false;
 
         private SteamVR_PlayArea playArea;
+        private SteamVR_Camera hmdCamera;
```
Comment (thestonefox) [inline Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs]: This is not used, best to delete it :)

Comment (sdouglas) [inline Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs]: Yikes, of course.

**Pull request merged**